vars/createUploadTestReportManifest.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to close GitHub issue across opensearch-project repositories.
+ @param Map args = [:] args A map of the following parameters
+ @param args.testManifest <required> - The path of the test manifest used
+ @param args.buildManifest <required> - The path of the build manifest of OpenSearch artifacts.
+ @param args.buildManifestDashboards <optional> - The path of the build manifest of OpenSearch Dashboards
+ @param args.testRunID <required> - Test run id of the test workflow being reported.
+ @param args.testType <required> - Type of the test workflow being reported.
+ @param args.componentName <Optional> - Components that workflow runs on.
+ */
+
+def call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+
+    def testID = args.testRunID;
+    def testType = args.testType;
+
+    def testManifest = lib.jenkins.TestManifest.new(readYaml(file: args.testManifest))
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.buildManifest))
+    def buildManifestDashboards = args.buildManifestDashboards ? lib.jenkins.BuildManifest.new(readYaml(file: args.buildManifestDashboards)) : null;
+
+    echo "Start Reporting workflow for test type: " + buildManifest.getDistribution()
+
+    String paths = generatePaths(testManifest, buildManifest, buildManifestDashboards)
+    echo "Paths: ${paths}"
+
+    String basePath = generateBasePaths(buildManifest)
+    echo "Base Path ${basePath}"
+
+    String component = args.componentName
+    echo "Component: ${component}"
+
+    String reportCommand =
+            [
+                    './report.sh',
+                    "${args.testManifest}",
+                    "--artifact-paths ${paths}",
+                    "--test-run-id ${testID}",
+                    "--test-type ${testType}",
+                    "--base-path ${basePath}",
+                    "--component ${component}",
```

Comment:
Since the component is an optional parameter, you can have a null check like [here](https://github.com/opensearch-project/opensearch-build-libraries/blob/main/vars/runBenchmarkTestScript.groovy#L106) and then change it to 
`isNullOrEmpty(args.componentName.toString()) ? "" : "--component ${args.componentName}",`

Reply:
Good suggestion. Updating here. Thanks!